### ISSUE TITLE
Styling improvements

### ISSUE THREAD

**Issue by cursorial** [wasp-lang/wasp#294]
**Is your feature request related to a problem? Please describe.**
A couple of issues to cover here for some general styling improvements:

- Trying to apply global font styles to the project is difficult, especially when attempting to use `styled-components`. I don't seem to have an easy way to access the root component of the app. I can't change `index.html` or `index.css` in the output either without it being over-written.
- Getting a gradient background color that fills the whole screen on my app is too difficult, again, I can't edit `index.css` without it being over-written when I next build.

Once you're in your React component, you have very little control of the context that it is in. I want to be able to define a global font family. I want to have full control of the `body` element on the page. I'd like to be able to plug into the `head` if I need to, what if I need a third party script in my `head`?

**Describe the solution you'd like**

Not sure if this is the best approach, but maybe some config for the boilerplate or the ability to 'eject' it so I can customise my `index.html` and `index.css` files to my liking.

**Comment by Martinsos** [comment]
@cursorial sorry for the late response, we were on vacation!

Awesome, I was hoping somebody would raise issues like these so we can figure out how to support what is needed.

Applying global font styles -> would would be the nicest way to do that? Would that be by adding code to the `<head>`? If so, we have `head` field that you can specify in `app` declaration. I am just looking at documentation and I realized we don't have this documented! We must add it to docs. But here is an example of its usage: https://github.com/wasp-lang/wasp/blob/master/waspc/examples/todoApp/todoApp.wasp .

As for global css, you can right now just define some css file and import it it any .js file that is used in the .wasp and that is it, it should work, however it certainly doesn't sound very elegant hm.

Regarding customizing context for React components (pages), I created this issue some time ago https://github.com/wasp-lang/wasp/issues/77 but we didn't get to it yet. I believe this would be on the right track?

Ok, to recap:
1. We have `app.head` property, for adding stuff to the `head`. It is somewhat rough but it works. Let me know if you have ideas how to make it nicer.
2. We certainly need more ways to customize "context" / body / ... . One idea is to allow people to provide their own top lvl React component or to customize existing one. Is there anything else you think would be useful, any specific ideas, how would you go about it?

**Comment by michael-curry** [comment]
> Applying global font styles -> would would be the nicest way to do that? Would that be by adding code to the <head>? If so, we have head field that you can specify in app declaration. I am just looking at documentation and I realized we don't have this documented! We must add it to docs. But here is an example of its usage: https://github.com/wasp-lang/wasp/blob/master/waspc/examples/todoApp/todoApp.wasp .

This is great, exactly the type of thing I was looking for. Would be good to get this into documentation for sure!

> As for global css, you can right now just define some css file and import it it any .js file that is used in the .wasp and that is it, it should work, however it certainly doesn't sound very elegant hm.

I think the `head` property should solve this problem.

> Regarding customizing context for React components (pages), I created this issue some time ago #77 but we didn't get to it yet. I believe this would be on the right track?

Looks like #77 will solve this problem quite nicely!

Thanks for the feedback on this, super helpful :)

**Comment by infomiho** [comment]
Users have been successful in using custom fonts via `@fontsource`, using the `client.rootComponent` to use different UI frameworks, using Tailwind etc. 

So I feel confident to close this issue.
